packages/core/src/client.ts

```diff
@@ -244,6 +246,19 @@ export class Client extends AsyncEventEmitter<MappedEvents> {
 
 		let timer: NodeJS.Timeout | undefined = createTimer(controller, timeout);
 
+		const onRatelimit = ({ data }: ToEventProps<GatewayRateLimitedDispatchData>) => {
+			controller.abort(new GatewayRateLimitError(data, options));
+		};
```

Comment:
Pretty sure this listener is wrong, you should check for the data (nonce, etc)

Reply:
on accident, good catch